Code/Source/RobotImporter/URDF/CollidersMaker.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "UrdfParser.h"
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/parallel/mutex.h>
+
+namespace ROS2
+{
+    //! Populates a given entity with all the contents of the <collider> tag in robot description.
+    class CollidersMaker
+    {
+    public:
+        CollidersMaker(const AZStd::string& modelPath);
+        ~CollidersMaker();
+
+        //! Builds .pxmeshes for every collider in link collider mesh.
+        //! @param link A parsed URDF tree link node which could hold information about colliders.
+        void BuildColliders(urdf::LinkSharedPtr link);
+        //! Add zero, one or many collider elements (depending on link content).
+        //! @param link A parsed URDF tree link node which could hold information about colliders.
+        //! @param entityId A non-active entity which will be affected.
+        void AddColliders(urdf::LinkSharedPtr link, AZ::EntityId entityId);
+        AZStd::vector<AZStd::string> m_meshesToBuild;
+        AZStd::mutex m_buildMutex;
```

Comment:
These variables should be private.

Reply:
They are in use outside of the class currently, but I see that it is due to a suboptimal design. URDFPrefabMaker will call a self-contained member function of CollidersMaker instead of a current solution which violates encapsulation. This will be done in the second commit.